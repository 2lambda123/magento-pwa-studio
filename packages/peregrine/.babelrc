{
  "plugins": [
    "transform-object-rest-spread",
    "transform-class-properties",
    ["transform-react-jsx", { "pragma": "createElement" }],
    ["transform-runtime", {
      "helpers": true,
      "polyfill": false, // polyfills will be handled by preset-env
      "regenerator": false
    }]
  ],
  "presets": [
    ["env", {
      "targets": {
        "browsers": ["> 5%"]
      },
      "modules": false
    }]
  ],
  "env": {
    "test": {
      "plugins": [
        "transform-es2015-modules-commonjs",
        "transform-object-rest-spread",
        "transform-class-properties",
        ["transform-react-jsx", { "pragma": "createElement" }]
      ]
    }
  }
}
