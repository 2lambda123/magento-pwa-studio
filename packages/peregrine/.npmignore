.eslintrc.js
.storybook
/lib/**/__{docs,helpers,mocks,stories,tests}__/**
/scripts
