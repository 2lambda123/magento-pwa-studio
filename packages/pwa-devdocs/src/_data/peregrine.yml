title: Peregrine
entries:
  - label: Overview
    url: /peregrine/

  - label: Storybook UI testing
    url: /peregrine/testing/storybook/

  - label: Utility modules
    entries:

      - label: REST API client
        url: /peregrine/reference/rest-api-client/

  - label: React components
    entries:

      - label: ContainerChild
        url: /peregrine/reference/container-child/

      - label: Item
        url: /peregrine/reference/item/

      - label: Items
        url: /peregrine/reference/items/

      - label: List
        url: /peregrine/reference/list/

      - label: Router
        url: /peregrine/reference/router/
