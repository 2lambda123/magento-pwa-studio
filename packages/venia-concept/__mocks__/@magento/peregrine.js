import React from 'react';
import { BrowserPersistence } from './util/simplePersistence';

const peregrine = jest.requireActual('../../../peregrine/src');

// re-exports
const RestApi = {
    Magento2: {
        request: jest.fn()
    }
};

const Util = { BrowserPersistence };

// hooks
const useApolloContext = jest.fn(peregrine.useApolloContext);
const useDocumentListener = jest.fn(peregrine.useDocumentListener);
const useDropdown = jest.fn(peregrine.useDropdown);
const useQuery = jest.fn(peregrine.useQuery);
const useQueryResult = jest.fn(peregrine.useQueryResult);
const useSearchParam = jest.fn(peregrine.useSearchParam);

// components

/**
 * the Price component from @magento/peregrine
 * has browser-specific functionality and cannot
 * currently by rendered in the test environment
 */
const Price = jest.fn().mockReturnValue(<div />);

module.exports = {
    ...peregrine,
    Price,
    RestApi,
    Util,
    useApolloContext,
    useDocumentListener,
    useDropdown,
    useQuery,
    useQueryResult,
    useSearchParam
};
