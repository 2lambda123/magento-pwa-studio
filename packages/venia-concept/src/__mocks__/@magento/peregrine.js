import { createElement } from 'react';

/**
 * the Price component from @magento/peregrine
 * has browser-specific functionality and cannot
 * currently by rendered in the test environment
 */
export const Price = () => <div />;
