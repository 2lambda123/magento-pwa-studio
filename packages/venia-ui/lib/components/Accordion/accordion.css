.root {
    border: 1px solid rgb(var(--venia-global-color-border));
    border-radius: 0.5em;
}
