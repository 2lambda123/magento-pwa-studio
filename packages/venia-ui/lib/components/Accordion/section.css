.root {
    border-bottom: 1px solid rgb(var(--venia-border));
}

.root:last-of-type {
    border-bottom-width: 0;
}

.contents_container {
    padding-left: 2em;
}

.title {
}

.title_container {
    cursor: pointer;
    display: flex;
    justify-content: space-between;
    padding: 2em;
    width: 100%;
}

.title_container_open {
    composes: title_container;
    border-bottom: solid 1px rgb(var(--venia-border));
}
