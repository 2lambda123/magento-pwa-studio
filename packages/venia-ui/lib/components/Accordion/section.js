import React, { useCallback } from 'react';
import { ChevronDown as ArrowDown, ChevronUp as ArrowUp } from 'react-feather';

import { useAccordionContext } from './accordion';
import Icon from '../Icon';

import { mergeClasses } from '../../classify';
import defaultClasses from './section.css';

const Section = props => {
    const { children, id, title } = props;

    const { handleSectionToggle, openSectionIds } = useAccordionContext();

    const handleSectionToggleWithId = useCallback(
        () => handleSectionToggle(id),
        [handleSectionToggle, id]
    );

    const isOpen = openSectionIds.has(id);
    const contents = isOpen ? children : null;
    const titleIconSrc = isOpen ? ArrowUp : ArrowDown;
    const titleIcon = <Icon src={titleIconSrc} />;

    const classes = mergeClasses(defaultClasses, props.classes);
    const titleContainerClass = isOpen
        ? classes.title_container_open
        : classes.title_container;

    return (
        <div className={classes.root}>
            <button
                className={titleContainerClass}
                onClick={handleSectionToggleWithId}
            >
                <div className={classes.title}>{title}</div>
                {titleIcon}
            </button>
            <div className={classes.contents_container}>{contents}</div>
        </div>
    );
};

export default Section;
