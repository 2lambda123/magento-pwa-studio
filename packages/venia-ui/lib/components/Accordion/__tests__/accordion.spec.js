import React from 'react';
import { createTestInstance } from '@magento/peregrine';

import Accordion from '../accordion';
import Section from '../section';

test('it renders a closed Section correctly', () => {
    // Act.
    const instance = createTestInstance(
        <Accordion>
            <Section title="The Section Title is always visible">
                This section is closed. You should not see this content.
            </Section>
        </Accordion>
    );

    // Assert.
    expect(instance.toJSON()).toMatchSnapshot();
});

test('it renders an open Section correctly', () => {
    // Act.
    const instance = createTestInstance(
        <Accordion>
            <Section title="The Section Title is always visible" isOpen={true}>
                This section is open. You should see this content.
            </Section>
        </Accordion>
    );

    // Assert.
    expect(instance.toJSON()).toMatchSnapshot();
});
