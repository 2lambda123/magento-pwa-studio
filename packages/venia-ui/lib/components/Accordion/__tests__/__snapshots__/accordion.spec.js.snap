// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a closed Section correctly 1`] = `
<div>
  <div>
    <button
      onClick={[Function]}
    >
      <div>
        The Section Title is always visible
      </div>
      <span>
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="6 9 12 15 18 9"
          />
        </svg>
      </span>
    </button>
    <div />
  </div>
</div>
`;

exports[`it renders an open Section correctly 1`] = `
<div>
  <div>
    <button
      onClick={[Function]}
    >
      <div>
        The Section Title is always visible
      </div>
      <span>
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="18 15 12 9 6 15"
          />
        </svg>
      </span>
    </button>
    <div>
      This section is open. You should see this content.
    </div>
  </div>
</div>
`;
